Society for Neuroscience Abstract 2016.js

```diff
@@ -0,0 +1,446 @@
+
+supported years: four digits after /pp8/ in the url specifies a year
+4376 ... 2017
+4071 ... 2016
+
+Ideally "target" should be as below to support "multiple" entries
+^https?://www\.abstractsonline\.com/pp8/#!/\d+/(presentation|presentations|sessions|participants)/
+
+If you find a failure or error, please let me know:
+Dr Kouichi C. Nakamura
+kouichi.c.nakamura@gmail.com
+*/
+
+
+function detectWeb(doc, url) {
+
+	if (url.search(/\/presentation\/\d+/) != -1) {
+		return 'conferencePaper';
+	} //else if (url.search(/(presentations|session|participant)\/[^\\]+\/\d+/) != -1) {
+	//	return 'multiple'; // does not work
+	//}
+}
+
+function doWeb(doc, url) {
+
+	if (detectWeb(doc,url) == 'conferencePaper') {
+		var id = [];
+		id[0] = url.match(/\d+$/)[0];
+		var baseurl = url.replace(/\d+$/,"");
+
+		scrape(id,baseurl);
+
+	} else if (detectWeb(doc,url) == 'multiple') {
+		//TODO does not work properly becuase the page HTML is dynamically generated and
+		// ZU.processDocuments() cannot obtain the actual page content from URL.
+
+		var items = {};
+
+		var bodyTitle = ZU.xpath(doc,'//h4[@class="name"]//span[@class="bodyTitle"]');
+
+		for (var i = 0; i < bodyTitle.length; i++) {
+			var item = new Zotero.Item("conferencePaper");
+			item.title = bodyTitle[i].innerText.replace(/^\d+\.\d+\.\s/,"");
+
+			item.id = ZU.xpath(doc,'//h4[@class="name"]/@data-id')[i].value;
+
+			items[item.id] = item.title;
+		}
+		//Zotero.debug(items);
+
+		Zotero.selectItems(items, function(selectedItems){
+
+		if(!selectedItems) return true;
+
+			var abstractIDs = [];
+			for (var i in selectedItems) {
+				abstractIDs.push(i);
+				//ZU.processDocuments(i, scrape);
+			}
+
+			var baseurl = url.replace(/presentations.+$|sessions.+$|participants.+$/,"presntation/");
+			Zotero.debug(baseurl);
+			scrape(abstractIDs,baseurl);
+		});
+
+	}
+}
+
+
+
+
+function scrape(ids,baseurl){
+
+Zotero.debug(ids);
+
+for (k = 0; k < ids.length; k++) {
+
+	var newurl = baseurl + ids[k];
+	Zotero.debug(newurl);
+
+	ZU.processDocuments(newurl, function (doc) {
+		// NOTE doGet()  html does not contain the actual contents,
+		// object is HTTPreqest object, use processDocuments
+		//
+		// Zotero.debug(doc); //object HTMLDocument
+		// Zotero.debug(doc.URL);
+
+		var item = new Zotero.Item("conferencePaper");
+
+		// pages, title
+		var color_primary = ZU.xpath(doc,'//h1[@class="color-primary"]');
+
+		//TODO when multiple doc does not contain the contents
+
+		// Zotero.debug(color_primary);
+		//TODO does not work for mulitiple
+
+		// var m1 = color_primary[0].innerHTML.match(/^\s*(\d+\.\d+)\n\s*\/\s(\w+\d+)\n\s*\-\s/);
+
+		// 282.18/B8, 1, 3.02
+		var m1 = color_primary[0].innerHTML.match(
+			/^\s*\n\s*(\d+)(\.(\d+))?\n(\s*\/\s([\w\/\d]+)\n)?\s*-\s/);
+
+		//Zotero.debug(m1);
+
+		var title;
+		if (m1 != null ) {
+			if (m1[3] != undefined) {
+				if (m1[5] != undefined) {
+					item.pages = m1[1] + "." + m1[3] + "/" + m1[5];
```

Comment:
Here I'm trying to capture presentation ID in the format of `282.18 / B8`, `3.02` or `8`. But looking at presentation IDs used by other conferences, the format is also conference- and presentation-type-dependent. Making this translator compatible with all other abstracts in abstractonline.com will require further adjustments of this part.

Reply:
Detection of presentation id has been simplified.

```javascript
var m1 = h1color_primary[0].innerHTML.match(/^[.\n\s\d\w\/\-]+(?=\s-\s)/);
m1 = m1[0].replace(/\s|\n/g,"");
item.pages = m1;
```

Reply:
As a general comment without looking into the details here: After a `match` you should usually test that the result this is not `null` and only in this case continue further, i.e. here
```js
var m1 = h1color_primary[0].innerHTML.match(/^[.\n\s\d\w\/\-]+(?=\s-\s)/);
if (m1) {
    item.pages = m1[0].replace(/\s|\n/g,"");
}
```

Reply:
Thanks!